### DISCUSSION TITLE
Using spatie/laravel-translatable with Livewire

### DISCUSSION THREAD

**Discussion by gizburdt** [livewire/livewire#2217] [1. Help] [answered]
I'm trying to use spatie/laravel-translatable in combination with Livewire and its model binding.
The title of the Document Model is translatable.

Livewire Component:
```
class Document extends Component
{
    public DocumentModel $document;

    protected $rules = [
        'document.title' => 'required|string|min:4',
    ];

    public function mount(DocumentModel $document)
    {
        $this->document = $document;
    }
}
```

View:
```
<input type="text" wire:model="document.title">
```

The value of the input is `[object Object]`, which is kind of logical, since the value of title is json/array.
Normally spatie/laravel-translatable will fix this with an accessor on `->title`, but I think it's different with Livewire.
Do you maybe know a solution or the place I need to look to get this to work?

Thanks :)

**Comment by joshhanley** [comment, marked as answer]
@gizburdt It might be easier not to bind to the model directly, instead create a title property and bind the input to that, then in your save/update method, you can add any appropriate translations and save the title property to your model.

Hope this helps!

**Reply by gizburdt** [reply]
Thanks! Thought of that too, but I hoped there was a way to automatically bind it to the model. Will try it!

Thanks again :)

**Reply by joshhanley** [reply]
No worries!

The only way that would work is if you can access the translations through dot notation, e.g. document.title.en to edit the english version. 

If you can access and modify that value like that, then you can bind a Livewire input to it using, `wire:model="document.title.en"`. 

But I'm unfamiliar with the package, so I don't know if they support accessing and mutation translations using dot notation.

**Reply by gizburdt** [reply]
Tried that too, but didn't work. Your first suggestion is now working in my application :)

**Reply by cjmellor** [reply]
Sorry to bring up an old topic, but it seems to be the only thing out there for this topic.

Do you have a code example of how you'd achieve this? I'm wanting to get the same results but struggling to achieve it.

Thanks :)
